File: packages/go/canonicalpath/posix.go
package canonicalpath

import "strings"

// ToPOSIX serializes a canonical POSIX or relative path for POSIX consumers.
func ToPOSIX(path Path) (string, error) {
	value := string(path)
	if strings.ContainsRune(value, '\x00') {
		return "", newError(ErrNULByte, "path contains NUL")
	}
	if hasDriveRoot(value) {
		return "", newError(ErrInvalidPath, "win32 drive paths require an explicit host mapping such as ToWSL")
	}
	if strings.Contains(value, "\\") {
		return "", newError(ErrInvalidPath, "canonical paths must use slash separators")
	}
	return value, nil
}

File: packages/ts/src/canonicalpath/posix.ts
import type { CanonicalPath } from "./types.js";
import { pathError } from "./errors.js";
import { hasDriveRoot } from "./internal.js";

export function toPOSIX(canonical: CanonicalPath): string {
  if (canonical.includes("\0")) throw pathError("ERR_NUL_BYTE", "path contains NUL");
  if (hasDriveRoot(canonical)) {
    throw pathError("ERR_INVALID_PATH", "win32 drive paths require an explicit host mapping such as toWSL");
  }
  if (canonical.includes("\\")) throw pathError("ERR_INVALID_PATH", "canonical paths must use slash separators");
  return canonical;
}
